### PR TITLE
cdi GF porting kit support for CDI TCK 4.0.

### PR DESCRIPTION
**Pull request by gurunrao**
cdi GF porting kit support for CDI TCK 4.0 and GF 7. 
Signed-off-by: gurunandan.rao@oracle.com <gurunandan.rao@oracle.com>

Full pass for GF 7 with CDI TCK 4.0 is expected post resolution of https://github.com/eclipse-ee4j/glassfish/issues/23872

CI run: https://ci.eclipse.org/jakartaee-tck/job/guru/job/cditck-porting-guru/job/jakartaee-10/82/

### CODE REVIEW

File: docker/run_cditck.sh [hunk 1]
```diff
@@ -160,10 +174,10 @@ if [[ -f ${REPORT}/cdi-$VER/test-report.html ]];then
 fi
 
 mv ${REPORT}/cdi-$VER/TEST-TestSuite.xml  ${REPORT}/cdi-$VER/cditck-$VER-junit-report.xml
-sed -i 's/name=\"TestSuite\"/name="cditck-3.0"/g' ${REPORT}/cdi-$VER/cditck-$VER-junit-report.xml
+sed -i 's/name=\"TestSuite\"/name="cditck-4.0"/g' ${REPORT}/cdi-$VER/cditck-$VER-junit-report.xml
 # Create Junit formated file for sigtests
 echo '<?xml version="1.0" encoding="UTF-8" ?>' > $REPORT/cdi-$VER-sig/cdi-$VER-sig-junit-report.xml
-echo '<testsuite tests="TOTAL" failures="FAILED" name="cdi-3.0.0-sig" time="0" errors="0" skipped="0">' >> $REPORT/cdi-$VER-sig/cdi-$VER-sig-junit-report.xml
+echo '<testsuite tests="TOTAL" failures="FAILED" name="cdi-$VER-sig" time="0" errors="0" skipped="0">' >> $REPORT/cdi-$VER-sig/cdi-$VER-sig-junit-report.xml
```
Comment (scottmarlow) [inline docker/run_cditck.sh:180]: Is cdi-$VER-sig-junit-report.xml updated during the TCK run?

Comment (scottmarlow) [reply docker/run_cditck.sh:180]: Also see https://github.com/eclipse-ee4j/cdi-tck/blob/master/doc/reference/src/main/asciidoc/sigtest.asciidoc regarding running of signature tests.

Comment (gurunrao) [reply docker/run_cditck.sh:180]: > Also see https://github.com/eclipse-ee4j/cdi-tck/blob/master/doc/reference/src/main/asciidoc/sigtest.asciidoc regarding running of signature tests for alternative to adding sigtest.jar (e.g. use maven to download it).

 PR changes are limited to GF7 support with CDI TCK 4.0.0, will raise an enhancement issue for sigtest jars.

Comment (gurunrao) [reply docker/run_cditck.sh:180]: > Is cdi-$VER-sig-junit-report.xml updated during the TCK run?

 cdi-$VER-sig-junit-report.xml is updated during tck run.

**Review by dmatej — APPROVED** [review]

**Pull request merged**